c7n/resources/s3control.py

```diff
@@ -95,3 +97,160 @@ class resource_type(TypeInfo):
         permission_prefix = 's3'
 
     source_mapping = {'describe': MultiRegionAccessPointDescribe}
+
+
+class StorageLensDescribe(DescribeSource):
+    def get_query_params(self, query_params):
+        query_params = query_params or {}
+        query_params['AccountId'] = self.manager.config.account_id
+        return query_params
+
+    def augment(self, resources):
+        client = local_session(self.manager.session_factory).client('s3control')
+        results = []
+        for r in resources:
+            arn = Arn.parse(r['StorageLensArn'])
+            storage_lens_configuration = client \
+                .get_storage_lens_configuration(AccountId=arn.account_id, ConfigId=r['Id'])
+            storage_lens_configuration.pop('ResponseMetadata', None)
+            tags = client \
+                .get_storage_lens_configuration_tagging(AccountId=arn.account_id, ConfigId=r['Id'])
+            storage_lens_configuration['Tags'] = tags['Tags']
+            results.append(storage_lens_configuration)
+        return results
+
+
+@resources.register('s3-storage-lens')
+class StorageLens(QueryResourceManager):
+    class resource_type(TypeInfo):
+        service = 's3control'
+        id = name = 'Id'
+        enum_spec = ('list_storage_lens_configurations', 'StorageLensConfigurationList', None)
+        arn = 'StorageLensArn'
+        arn_service = 's3'
+        arn_type = 'storage-lens'
+        cfn_type = 'AWS::S3::StorageLens'
+        permission_prefix = 's3'
+
+    source_mapping = {'describe': StorageLensDescribe}
+
+
+@StorageLens.action_registry.register('tag')
+class TagStorageLens(Tag):
+    """Create tags on s3 storage lens
+
+    :example:
+
+    .. code-block:: yaml
+
+        policies:
+            - name: s3-storage-lens-tag
+              resource: aws.s3-storage-lens
+              actions:
+                - type: tag
+                  key: test
+                  value: storagelens
+    """
+    permissions = ('s3:GetStorageLensConfigurationTagging',
+                   's3:PutStorageLensConfigurationTagging',)
+
+    def process_resource_set(self, client, resources, add_tags):
+        accountId=self.manager.config.account_id
+        for r in resources:
+            configId=r['StorageLensConfiguration']['Id']
```

Comment:
nit, its unfortunate that ruff doesn't pick this up, but please put spaces between operators re `x = 1`

Reply:
sure, it's updated.